File: src/main/java/com/metamong/movie/service/ReviewService.java
package com.metamong.movie.service;

import com.metamong.movie.dto.ReviewRequestDto;
import com.metamong.movie.dto.ReviewResponseDto;
import com.metamong.movie.entity.Review;
import com.metamong.movie.repository.ReviewRepository;
import org.springframework.stereotype.Component;

import java.util.List;

@Component

public class ReviewService{
    private final ReviewRepository reviewRepository;
    public ReviewService(ReviewRepository reviewRepository) {
        this.reviewRepository = reviewRepository;
    }

    /**
     * CREATE
     * @param reviewRequestDto
     * @return
     */
    public ReviewResponseDto createReview(ReviewRequestDto reviewRequestDto) {
        Review review = new Review(reviewRequestDto);

        Review saveReview = reviewRepository.save(review);

        ReviewResponseDto reviewResponseDto = new ReviewResponseDto(saveReview);

        return reviewResponseDto;
    }

    /**
     * READ
     * @return
     */
    public List<ReviewResponseDto> getReviews() {
        List<ReviewResponseDto> reviewResponseDtos = reviewRepository.getreivews();
        return reviewResponseDtos;
    }

    /**
     * UPDATE
     * @param id
     * @param reviewRequestDto
     * @return
     */
    public Long updateReview(Long id, ReviewRequestDto reviewRequestDto) {
        Review review = reviewRepository.findById(id);
        if(review != null){
            reviewRepository.update(id, reviewRequestDto);

            return id;
        }else{
            throw new IllegalArgumentException("선택한 리뷰는 존재하지 않습니다");
        }
    }

    /**
     * DELETE
     * @param id
     * @return
     */
    public Long deleteReview(Long id) {
        Review review = reviewRepository.findById(id);
        if(review !=null){
            reviewRepository.delete(id);

            return id;
        }else{
            throw new IllegalArgumentException("선택한 리뷰는 존재하지 않습니다");
        }
    }
}

File: src/res/review/service/review.service.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { ReviewRequestDto } from '../dto/review-request.dto';
import { ReviewResponseDto } from '../dto/review-response.dto';
import { Review } from '../entity/review.entity';
import { ReviewRepository } from '../repository/review.repository';

@Injectable()
export class ReviewService {
  constructor(
    private readonly reviewRepository: ReviewRepository,
  ) {}

  async createReview(
    reviewRequestDto: ReviewRequestDto,
  ): Promise<ReviewResponseDto> {
    const review = new Review(reviewRequestDto);
    const savedReview = await this.reviewRepository.save(review);
    return new ReviewResponseDto(savedReview.id, savedReview.user.username, savedReview.content);
  }

  async getReviews(): Promise<ReviewResponseDto[]> {
    return await this.reviewRepository.getReviews();
  }

  async updateReview(
    id: number,
    reviewRequestDto: ReviewRequestDto,
  ): Promise<number> {
    await this.reviewRepository.update(id, reviewRequestDto);
    return id;
  }

  async deleteReview(id: number): Promise<number> {
    await this.reviewRepository.delete(id);
    return id;
  }
}
